api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupPNCRegister.java

```diff
@@ -0,0 +1,243 @@
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "PNCRegister.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8-10,dataset:PNC");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public String getUuid() {
+        return "489baf0e-3ee1-11e7-a1d1-507b9dc4c741";
+    }
+
+    @Override
+    public String getName() {
+        return "Integrated Postnatal Register";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Contains mothers information after delivery";
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.addParameters(getParameters());
+        rd.addDataSetDefinition("PNC", Mapped.mapStraightThrough(dataSetDefinition()));
+        return rd;
+    }
+
+    @Override
+    public String getVersion() {
+        return "0.1";
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        List<Parameter> l = new ArrayList<Parameter>();
+        l.add(df.getStartDateParameter());
+        l.add(df.getEndDateParameter());
+        return l;
+    }
+
+    private DataSetDefinition dataSetDefinition() {
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+        dsd.setName("PNC");
+        dsd.addParameters(getParameters());
+        dsd.addRowFilter(Cohorts.genderAndHasAncEncounter(true, false, "fa6f3ff5-b784-43fb-ab35-a08ab7dbf074"), "startDate=${startDate},endDate=${endDate}");
+
+        PersonAttributeType attribute = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
+        PatientIdentifierType preARTNo = MetadataUtils.existing(PatientIdentifierType.class, "e1731641-30ab-102d-86b0-7a5022ba4115");
+        DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+        DataDefinition identifierDefPre = new ConvertedPatientDataDefinition("identifier", new PatientIdentifierDataDefinition(preARTNo.getName(), preARTNo), identifierFormatter);
+
+
+        //start adding columns
+        dsd.addColumn("Serial No", sdd.definition("Serial No",  getConcept("1646AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+        dsd.addColumn("Client No", sdd.definition("Client No", getConcept("ef1f4c7a-2b90-4412-83bb-87ae8094ce4c")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter()         );
+        dsd.addColumn("Mother Name", new PreferredNameDataDefinition(), (String) null);
+        dsd.addColumn("Village+Parish", villageParish(), "onDate=${endDate}", new CalculationResultDataConverter());
```

Comment:
@ssmusoke am using this CalculationResultDataConverter as a general data converter for everything that is being returned by a calculation, it will be used in other registers in future. I will keep it as it is